src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -255,43 +254,108 @@ protected function addIns(array $inFields, array $inValues)
             $this->queryBuilder->setParameter($betweenField->getName() . '2', $values[1]);
         }
 
-        $this->queryBuilder->andWhere('(' . implode(' AND ', $betweenParts) . ')');
+        $betweenString = $this->createQueryString($betweenParts, strlen($firstConjunction) + 2);
+        if (strtoupper($firstConjunction) === self::OR_CONJUNCTION) {
+            $this->queryBuilder->orWhere('(' . $betweenString . ')');
+        } else {
+            $this->queryBuilder->andWhere('(' . $betweenString . ')');
+        }
+    }
+
+    /**
+     * Concats array to string
+     * @param array $parts
+     * @param int $skipLetters
+     * @param string $glue
+     * @return string
+     */
+    protected function createQueryString(array $parts, $skipLetters, $glue = '')
+    {
+        $tmp = implode($glue, $parts);
+        return substr($tmp, $skipLetters);
     }
 
     /**
      * sets where statement
      * @param array $whereFields
      * @param array $whereValues
-     * @param string $comparator
+     * @param array $whereComparators
+     * @param array $whereConjunctions
+     * @internal param string $comparator
      */
-    protected function addWheres(array $whereFields, array $whereValues, $comparator = self::WHERE_COMPARATOR_EQUAL)
-    {
+    protected function addWheres(
+        array $whereFields,
+        array $whereValues,
+        array $whereComparators,
+        array $whereConjunctions
+    ) {
         $whereParts = array();
+        $firstConjunction = null;
+
         foreach ($whereFields as $whereField) {
+            $conjunction = ' ' . $whereConjunctions[$whereField->getName()] . ' ';
             $value = $whereValues[$whereField->getName()];
+            $comparator = $whereComparators[$whereField->getName()];
+            $whereParts[] = $this->createWherePart($value, $whereField, $conjunction, $comparator);
 
-            if ($value === null) {
-                $whereParts[] = $whereField->getSelect() . ' ' . $this->convertNullComparator($comparator);
+            if (!$firstConjunction) {
+                $firstConjunction = $whereConjunctions[$whereField->getName()];
+            }
+        }
+
+        $whereString = $this->createQueryString($whereParts, strlen($firstConjunction) + 2);
+        if (strtoupper($firstConjunction) === self::OR_CONJUNCTION) {
+            $this->queryBuilder->orWhere('(' . $whereString . ')');
+        } else {
+            $this->queryBuilder->andWhere('(' . $whereString . ')');
+        }
+    }
+
+    /**
+     * Creates a partial where statement
+     * @param $value
+     * @param $whereField
+     * @param $conjunction
+     * @param $comparator
+     * @return string
+     */
+    protected function createWherePart($value, $whereField, $conjunction, $comparator)
+    {
+        if ($value === null) {
+
+            return $conjunction . $whereField->getSelect() . ' ' . $this->convertNullComparator($comparator);
+        } else {
```

Comment:
You don't need `else` here.

````
if ($value === null) {
    return //
}

// do something else
````

You should try always to avoid the use of `else`, it is frequently not necessary.